cli/src/semdep/parse_lockfile.py

```diff
@@ -151,38 +152,49 @@ def parse_package_lock(
         manifest_deps = manifest["dependencies"] if "dependencies" in manifest else {}
     else:
         manifest_deps = None
-    for dep, dep_blob in deps.items():
-        version = dep_blob.get("version")
-        if not version:
-            logger.info(f"no version for dependency: {dep}")
-            continue
-        try:
-            Version(version)
-        # Version was a github commit
-        except InvalidVersion:
-            logger.info(f"no version for dependency: {dep}")
-            continue
-        line_number = dep_blob.get("line_number")
-        if manifest_deps:
-            transitivity = (
-                Transitivity(Direct())
-                if dep in manifest_deps
-                else Transitivity(Transitive())
+
+    def parse_deps(deps: Dict[str, Any], nested: bool) -> Iterator[FoundDependency]:
+        for dep, dep_blob in deps.items():
+            version = dep_blob.get("version")
+            if not version:
+                logger.info(f"no version for dependency: {dep}")
+                continue
+            try:
+                Version(version)
+            # Version was a github commit
+            except InvalidVersion:
+                logger.info(f"no version for dependency: {dep}")
+                continue
+            line_number = dep_blob.get("line_number")
+            if nested:
+                # Nested dependencies are always transitive
+                transitivity = Transitivity(Transitive())
+            elif manifest_deps:
+                transitivity = (
+                    Transitivity(Direct())
+                    if dep in manifest_deps
+                    else Transitivity(Transitive())
+                )
+            else:
+                transitivity = Transitivity(Unknown())
+
+            resolved_url = dep_blob.get("resolved")
+            integrity = dep_blob.get("integrity")
+            yield FoundDependency(
+                package=dep,
+                version=version,
+                ecosystem=Ecosystem(Npm()),
+                allowed_hashes=extract_npm_lockfile_hash(integrity)
+                if integrity
+                else {},
+                resolved_url=resolved_url,
+                transitivity=transitivity,
+                line_number=int(line_number) + 1 if line_number else None,
             )
-        else:
-            transitivity = Transitivity(Unknown())
+            if nested_deps := dep_blob.get("dependencies"):
```

Comment:
Is this endlessly recursive or will package-lock.json only go one level down anyway

Reply:
I'm fairly sure it can be multiple levels but this is one of the things I want to test this week. I just want to get this in because I really don't want another message to the effect of "we're missing some unreachable findings" from a customer lol